test/transform.js

```diff
@@ -327,3 +327,28 @@ check(
 
   , 'gitlab.com'
 )
+
+// check that the asteriskFlag successfully uses '* ' instead of '- '
+check(    
+    [ '# My Module'
+    , 'Some text here'
+    , '## API'
+    , '### Method One'
+    , 'works like this'
+    , '### Method Two'
+    , '#### Main Usage'
+    , 'some main usage here'
+    ].join('\n')
+  , [ '**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*\n\n'
+    , '* [My Module](#my-module)\n'
+    ,   '  * [API](#api)\n'
+    ,     '    * [Method One](#method-one)\n'
+    ,     '    * [Method Two](#method-two)\n'
+    ,         '      * [Main Usage](#main-usage)\n\n\n'
+    ].join('')
+  , undefined
+  , undefined
+  , undefined
+  , undefined
+  , true // set asteriskFlag which is passed to transform() to true
+)
```

Comment:
Looks like your editor isn't configured to add newlines at the end of the file on save. See "[Why should text files end with a newline?][1]" for more information. 

[1]: http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline